cookies/accession/zotero.py

```diff
@@ -430,14 +429,11 @@ def __iter__(self):
     def next(self):
         next_entry = None
         while next_entry is None:
-            try:
-                next_entry = self.current_entries.next()
-            except (StopIteration, AttributeError):
-                try:
-                    self.current_class = self.classes.pop()
-                except IndexError:    # Out of classes.
-                    raise StopIteration()
-                self.current_entries = self._get_resources_nodes(self.current_class)
+            next_entry = self.current_entries.next()
+            for i in self.graph.objects(subject=next_entry):
```

Comment:
```i``` is a pretty bad variable name

Reply:
why don't you provide the predicate you're looking for here? there is no reason to iterate over all triples. in addition, if any of the other triples have an object 'attachment', the entry will be skipped although it might be a real one.

Reply:
also, just assume there is either no itemtype or one. if there are more than one, this would be an error anyways and the other ones can be disregarded.